dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/EncoderTest.java

```diff
@@ -0,0 +1,456 @@
+                          "method": "tools/call",
+                          "params": {
+                            "name": "testEncoderPriority",
+                            "arguments": {}
+                          }
+                        }
+                        """;
+
+        String response = client.callMCP(request);
+        // the object within the text field is expected to have the fields in lexicographical order after converting the object to JSON
+        // 3 backslashes, as it should look like \" in the response. So we need extra backslashes to escape the \ and to escape the "
+        String expectedResponseString = """
+                        {
+                          "id":"2",
+                          "jsonrpc":"2.0",
+                          "result": {
+                            "content": [
+                              {
+                                "type":"text",
+                                "text":"{\\\"hello\\\":\\\"Hello from HigerPriorityEncoder\\\"}"
+                              }
+                            ],
+                            "isError": false
+                          }
+                        }
+                        """;
+        JSONAssert.assertEquals(expectedResponseString, response, true);
+    }
+
+    @Test
+    public void testToolResponseEncoderSuccess() throws Exception {
+        String request = """
+                          {
+                          "jsonrpc": "2.0",
+                          "id": "2",
+                          "method": "tools/call",
+                          "params": {
+                            "name": "testToolResponseEncoder",
+                            "arguments": {
+                                   "isSuccessful": true
+                            }
+                          }
+                        }
+                        """;
+
+        String response = client.callMCP(request);
+        // the object within the text field is expected to have the fields in lexicographical order after converting the object to JSON
+        // 3 backslashes, as it should look like \" in the response. So we need extra backslashes to escape the \ and to escape the "
+        String expectedResponseString = """
+                        {
+                          "id":"2",
+                          "jsonrpc":"2.0",
+                          "result": {
+                            "content": [
+                              {
+                                "type":"text",
+                                "text":"[{\\"country\\":\\"England\\",\\"isCapital\\":true,\\"name\\":\\"London\\",\\"population\\":18000},{\\"country\\":\\"England\\",\\"isCapital\\":false,\\"name\\":\\"Machester\\",\\"population\\":8000}]"
+                              }
+                            ],
+                            "isError": false
+                          }
+                        }
+                        """;
```

Comment:
Shouldn't this create two separate text content objects in the response, rather than one which has a json encoded list of two items?

Based on https://github.com/OpenLiberty/open-liberty/pull/33358/files#diff-67721b506ba4d77820d520cec1484aa94555eaa88fb6b53f37dabde30bf86b6eR155-R157

I'm not immediately sure what's going on here given that the next test passes and shows that the encoder is being called.